### PR TITLE
[feature-725]: Remove k3s from karavictl

### PR DESCRIPTION
**Pull request by atye**
# Description

- Removes k3s calls from role and storage commands so all commands will connect to proxy-server
- Moved `addr` and `insecure` flag to root command
- Removes karavictl grpc tests and unnecessary test files

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/725 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Existing unit tests and manual tests.

### CODE REVIEW

File: cmd/karavictl/cmd/api/api.go [hunk 1]
```diff
@@ -312,8 +312,19 @@ func (c *client) DoAndGetResponseBody(
 // ParseJSONError parses the error from the proxy server
 func (c *client) ParseJSONError(r *http.Response) error {
 	jsonError := web.JSONError{}
-	if err := json.NewDecoder(r.Body).Decode(&jsonError); err != nil {
-		return err
+
+	b, err := io.ReadAll(r.Body)
```
Comment (alikdell) [inline cmd/karavictl/cmd/api/api.go:316]: I have this changes in my PR

Comment (atye) [reply cmd/karavictl/cmd/api/api.go]: Reverted to original and will let your PR handle it.

**Review by alikdell — APPROVED** [review]

**Review by IsaiasA1 — DISMISSED** [review]

**Review by alikdell — DISMISSED** [review]

**Review by shaynafinocchiaro — APPROVED** [review]

**Review by alikdell — APPROVED** [review]

File: cmd/karavictl/cmd/role_get.go [hunk 2]
```diff
@@ -56,6 +57,21 @@ func NewRoleGetCmd() *cobra.Command {
...
+			if err != nil {
+				reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
+			}
+
+			query := url.Values{
+				"name": []string{roleName},
+			}
+
+			var role pb.RoleGetResponse
+			err = client.Get(ctx, "/proxy/roles", nil, query, &role)
```
Comment (alikdell) [inline cmd/karavictl/cmd/role_get.go:70]: We have to provide admin token header in this request

**Review by alikdell — DISMISSED** [review]

**Review by IsaiasA1 — DISMISSED** [review]

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Review by alikdell — APPROVED** [review]

**Pull request merged**